### ISSUE TITLE
The test test-bad-surrogate assumes code-char can return a surrogate character

### ISSUE THREAD

**Issue by Yehouda** [y2q-actionman/cl-scsu#3]
The test **test-bad-surrogate** calls **code-char** on a code corresponding to a surrogate character, and then pass the result to **(setf schar)**. On LispWorks (and I think also CCL) **code-char** returns NIL for such a code, so the test gets an error in the **(setf schar)**.

LispWorks puts some effort in stoppig users from creating string containing surrogate characters, so there is no obvious way to do what the test tries to do. I suggest putting **(when (ignore-errors (code-char #xd800))** around the first part of the **test-bad-surrogate**.

**Comment by y2q-actionman** [comment]
I confirmed `code-char` for a bad character returns NIL, using LispWorks personal. (I used it for the first time!) 
I added a workaround you suggested. (commit 1f72ca008ec9b7af984575f4e233b77f72c48b3b)

**Comment by Yehouda** [comment]
In other projects the owner of the project normally closes the issue once they fixed it, but I see you don't. Is that because you expect me to close them?

As far as I am concerned, this and the previous two issues can be closed. 

I don't actually pull code from git directly, instead I test what there is in quicklisp, and I tested these changes by editing by hand the code that I got from quicklisp.  So I won't really test the changes in the project itself until the next release of quicklisp.
